tasks/static_quality_gates/lib/gates_lib.py

```diff
@@ -0,0 +1,139 @@
+        raise Exit(code=1, message=color_message(f"Couldn't find any file matching {glob_pattern}", "red"))
+    return package_paths[0]
+
+
+class GateMetricHandler:
+    def __init__(self, git_ref, bucket_branch):
+        self.metrics = {}
+        self.metadata = {}
+        self.git_ref = git_ref
+        self.bucket_branch = bucket_branch
+
+    def get_formated_metric(self, gate_name, metric_name):
+        return byte_to_string(self.metrics[gate_name][metric_name])
+
+    def register_metric(self, gate_name, metric_name, metric_value):
+        if self.metrics.get(gate_name, None) is None:
+            self.metrics[gate_name] = {}
+
+        self.metrics[gate_name][metric_name] = metric_value
+
+    def register_gate_tags(self, gate, **kwargs):
+        if self.metadata.get(gate, None) is None:
+            self.metadata[gate] = {}
+
+        for key in kwargs:
+            self.metadata[gate][key] = kwargs[key]
+
+    def _generate_series(self):
+        if not self.git_ref or not self.bucket_branch:
+            return None
+
+        series = []
+        timestamp = int(datetime.utcnow().timestamp())
+        for gate in self.metrics:
+            common_tags = [
+                f"git_ref:{self.git_ref}",
+                f"bucket_branch:{self.bucket_branch}",
+            ]
+
+            if self.metadata.get(gate, None) is None:
+                print(color_message(f"[WARN] gate {gate} doesn't have gate tags registered ! skipping...", "orange"))
+                continue
+
+            for tag in self.metadata[gate]:
+                common_tags.append(f"{tag}:{self.metadata[gate][tag]}")
+
+            series.append(
+                create_gauge(
+                    "datadog.agent.static_quality_gate.on_wire_size",
+                    timestamp,
+                    self.metrics[gate]["current_on_wire_size"],
+                    tags=common_tags,
+                    metric_origin=get_metric_origin(ORIGIN_PRODUCT, ORIGIN_CATEGORY, ORIGIN_SERVICE),
+                ),
+            )
+            series.append(
+                create_gauge(
+                    "datadog.agent.static_quality_gate.on_disk_size",
+                    timestamp,
+                    self.metrics[gate]["current_on_disk_size"],
+                    tags=common_tags,
+                    metric_origin=get_metric_origin(ORIGIN_PRODUCT, ORIGIN_CATEGORY, ORIGIN_SERVICE),
+                ),
+            )
+            series.append(
+                create_gauge(
+                    "datadog.agent.static_quality_gate.max_allowed_on_wire_size",
+                    timestamp,
+                    self.metrics[gate]["max_on_wire_size"],
+                    tags=common_tags,
+                    metric_origin=get_metric_origin(ORIGIN_PRODUCT, ORIGIN_CATEGORY, ORIGIN_SERVICE),
+                ),
+            )
+            series.append(
+                create_gauge(
+                    "datadog.agent.static_quality_gate.max_allowed_on_disk_size",
```

Comment:
Similar to my other comments around units, ensure that the correct metric unit is set in org2's metric-summary configuration for all of these.